### ISSUE TITLE
Fix the task API

### ISSUE THREAD

**Issue by irevoire** [meilisearch/meilisearch#3342]
Due to a misunderstanding in the spec of the tasks filter, the implementation was bugged and needs to be fixed.
See the following spec for the changes: https://github.com/meilisearch/specifications/pull/218
FYI: the bug is present is v0.30 as well.

### The bug:
The `*` was accepted on some parameters but not all the specified, and it was buggy. i.e. if you were sending `tasks/delete?uids=*`, it would refuse your filter, but `tasks/delete?uids=*&statuses=succeded` would work

### TLDR: The implementation
Now the `*` is accepted and handled as expected for the;
- uids
- statuses
- types
- indexUids
- beforeEnqueuedAt
- beforeStartedAt
- beforeFinishedAt
- afterEnqueuedAt
- afterStartedAt
- afterFinishedAt

Also, pinging @meilisearch/docs-team here because I noticed that you documented the `*` correctly for all the fields except the dates filters.

**Comment by irevoire** [comment]
Fixed by https://github.com/meilisearch/meilisearch/pull/3339
